contracts/FanDonation.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: agpl-3.0
+
+pragma solidity ^0.6.0;
+
+import {IERC20} from "../contracts/interfaces/IERC20.sol";
+import {SafeERC20} from "../contracts/interfaces/SafeERC20.sol";
+
+import {ILendingPool} from "../contracts/interfaces/ILendingPool.sol";
+import {ILendingPoolAddressesProvider} from "../contracts/interfaces/ILendingPoolAddressesProvider.sol";
+
+contract FanDonation {
+    using SafeERC20 for IERC20;
+
+    // Set the address of the AAVE protocol addresses provider. Should not change once deployed.
+    // Currently set to the LendingPoolAddressesProvider for Kovan (https://docs.aave.com/developers/deployed-contracts)
+    ILendingPoolAddressesProvider public constant addressesProvider = ILendingPoolAddressesProvider(0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5);
+    ILendingPool lendingPool = ILendingPool(addressesProvider.getLendingPool());
+
+    address asset = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48); // USDC
+    uint16 referralCode = 0;
+
+    // Signals to the front end whether or not the donation was successful
+    event DonationCompleteStatus(bool success);
+
+    function donate(uint256 amount, address _onBehalfOf) external {
+        IERC20(asset).safeApprove(address(lendingPool), amount);
```

Comment:
Will this appear as two transactions for the user? One to approve first, then the deposit transaction afterwards?

If not, we may need to separate these into two functions that the front-end will call? approve(asset) and donate(asset, amount, address)? I'm not 100% sure if this works within the same function automatically, but would be great if it does.

Do we want to require() that the fan has approved the ERC20 here, i.e. IERC20(asset).allowance(fan, thisContract) > 0, if we do separate this into two functions?

Reply:
Oops, just realized that safeApprove already checks the last part of my comment, so ignore that part!